src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+          block.pathObject.updateSelected(false);
+        }
+      });
+      Blockly.Events.setGroup(false);
+      blockSelection.clear();
+      duplicatedBlocks.forEach(function(block) {
+        if (block.id) {
+          blockSelection.add(block.id);
+          block.pathObject.updateSelected(true);
+        }
+      });
+    },
+    scopeType: Blockly.ContextMenuRegistry.ScopeType.BLOCK,
+    id: 'blockDuplicate',
+    weight: 1,
+  };
+  Blockly.ContextMenuRegistry.registry.register(duplicateOption);
+};
+
+/**
+ * Modification for context menu 'Comment' to be available for multiple blocks.
+ */
+const registerComment = function() {
+  const commentOption = {
+    displayText: function(scope) {
+      const selectedBlocksLength = blockSelection.size;
+      if (scope.block.getCommentIcon()) {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['REMOVE_COMMENT'];
+        } else {
+          return Blockly.Msg['REMOVE_COMMENT'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      }
+      if (selectedBlocksLength <= 1) {
+        return Blockly.Msg['ADD_COMMENT'];
+      } else {
+        return Blockly.Msg['ADD_COMMENT'] + ' (' +
+        selectedBlocksLength + ')';
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!Blockly.utils.userAgent.IE && !block.isInFlyout &&
+        block.workspace.options.comments && !block.isCollapsed() &&
+        block.isEditable()) {
+        return 'enabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const hasCommentIcon = scope.block.getCommentIcon();
+      const apply = function(block) {
+        if (block && commentOption.preconditionFn({
+          block: block,
+        }) === 'enabled') {
+          if (hasCommentIcon) {
+            block.setCommentText(null);
+          } else {
+            block.setCommentText('');
+          }
+        }
+      };
+      Blockly.Events.setGroup(true);
+      if (scope.block && blockSelection.size === 0) {
+        apply(scope.block);
+      }
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (block) {
+          apply(block);
+        }
+      });
+      Blockly.Events.setGroup(false);
+    },
+    scopeType: Blockly.ContextMenuRegistry.ScopeType.BLOCK,
+    id: 'blockComment',
+    weight: 2,
+  };
+  Blockly.ContextMenuRegistry.registry.register(commentOption);
+};
+
+/**
+ * Modification for context menu 'Inline' to be available for multiple blocks.
+ */
+const registerInline = function() {
+  const inlineOption = {
+    displayText: function(scope) {
+      const selectedBlocksLength = blockSelection.size;
+      if (scope.block.getInputsInline()) {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['EXTERNAL_INPUTS'];
+        } else {
+          return Blockly.Msg['EXTERNAL_INPUTS'] + ' (' +
+          selectedBlocksLength + ')';
```

Comment:
As with the `registerComment` option, I think this should try to provide an accurate count of how many blocks are being modified.